src/components/Layout/ContainerWithHeader.js

```diff
@@ -83,31 +85,32 @@ class ContainerWithHeader extends React.Component<Props, State> {
 
   componentDidMount() {
     const { navigation } = this.props;
-    this.focusSubscriptions = [
-      navigation.addListener('didFocus', this.setStatusBarStyleForView),
-      navigation.addListener('willBlur', this.resetStatusBarStyle),
-    ];
+    this.focusSubscription = navigation.addListener('didFocus', this.setStatusBarStyleForView);
   }
 
   componentWillUnmount() {
-    this.resetStatusBarStyle();
-    this.focusSubscriptions.forEach(sub => sub.remove());
+    this.focusSubscription.remove();
   }
 
+  getStatusBarColor = (themeType) => {
+    if (themeType === DARK_THEME) return 'light-content';
+    return 'dark-content';
+  };
+
   setStatusBarStyleForView = () => {
-    const { headerProps = {} } = this.props;
-    const { color } = headerProps;
-    let statusBarStyle = 'dark-content';
-    if (color && isColorDark(color)) {
-      statusBarStyle = 'light-content';
+    const { headerProps = {}, theme, backgroundColor } = this.props;
+    const { transparent, floating } = headerProps;
+    const themeType = getThemeType(theme);
+    let statusBarStyle = this.getStatusBarColor(themeType);
+
+    if ((!!transparent || !!floating) && backgroundColor) {
+      statusBarStyle = isColorDark(backgroundColor)
+        ? this.getStatusBarColor(DARK_THEME)
+        : this.getStatusBarColor(LIGHT_THEME);
```

Comment:
Shouldn't this match the same behaviour when setting status bar according to theme – dark theme sets light status bar?